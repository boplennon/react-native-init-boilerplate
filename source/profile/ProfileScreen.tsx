import * as React from 'react';
import { CrossFbLoginButton } from '../facebook/CrossFbLoginButton';
import styles, { TabIconSize } from '../styles';
import { View, Text } from 'react-native';
import { CrossTabStatelessComponent } from '../CrossNavigator';
import FontAwesome from 'react-native-vector-icons/FontAwesome';

export const ProfileScreen:CrossTabStatelessComponent = () => (
  <View style={[styles.container, styles.paddingDefault]}>
    <CrossFbLoginButton />
  </View>
);

ProfileScreen.navigationOptions = {
  tabBarIcon: <FontAwesome name="user" size={TabIconSize} color="white" />,
  tabBarLabel: <Text style={styles.tabBarLabel}>Home</Text>,
};

export default ProfileScreen;
