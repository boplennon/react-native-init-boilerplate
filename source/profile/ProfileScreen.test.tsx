import React from 'react';
import ProfileScreen from './ProfileScreen';
import TestRenderer from 'react-test-renderer';

function setup() {
    return TestRenderer.create(
        <ProfileScreen />
    );
}

describe('components', () => {
    describe('<ProfileScreen />', () => {
        it('renders correctly with defaults', () => {
            const wrapper = setup();
            expect(wrapper.toJSON()).toMatchSnapshot();
        });
    });
});