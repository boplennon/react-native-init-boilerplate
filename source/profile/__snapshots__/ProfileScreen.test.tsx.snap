// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components <ProfileScreen /> renders correctly with defaults 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "justifyContent": "space-between",
      },
      Object {
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 20,
      },
    ]
  }
>
  <Button
    onLoginFinished={[Function]}
    onLogoutFinished={[Function]}
    style={
      Object {
        "height": 45,
        "marginBottom": 10,
        "marginRight": 10,
        "marginTop": 10,
      }
    }
  />
</View>
`;
